### PR TITLE
audit: make audit_revision_and_version_scheme faster.

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
This is really, really slow at the moment for a few reasons:
- it goes through the list of revisions twice
- it checks many more revisions than it needs to
- it checks `devel` even when that's never used in homebrew/core

Even after these improvements it's still by far the slowest audit so am also making it a strict or new-formula audit.

before:
```
audit_postgresql:                   0.0000 sec
audit_gitlab_repository:            0.0000 sec
audit_bottle_spec:                  0.0000 sec
audit_github_repository:            0.0000 sec
audit_versioned_keg_only:           0.0000 sec
audit_style:                        0.0000 sec
audit_bottle_disabled:              0.0000 sec
audit_conflicts:                    0.0000 sec
audit_bitbucket_repository:         0.0000 sec
audit_homepage:                     0.0000 sec
audit_specs:                        0.0002 sec
audit_reverse_migration:            0.0004 sec
audit_prefix_has_contents:          0.0011 sec
audit_lines:                        0.0011 sec
audit_formula_name:                 0.0026 sec
audit_file:                         0.0068 sec
audit_deps:                         0.0125 sec
audit_text:                         0.0207 sec
audit_installed:                    0.1784 sec
audit_revision_and_version_scheme:  32.1020 sec
```

after:
```
audit_postgresql:                   0.0000 sec
audit_bottle_spec:                  0.0000 sec
audit_github_repository:            0.0000 sec
audit_versioned_keg_only:           0.0000 sec
audit_gitlab_repository:            0.0000 sec
audit_homepage:                     0.0000 sec
audit_bottle_disabled:              0.0000 sec
audit_conflicts:                    0.0000 sec
audit_style:                        0.0000 sec
audit_bitbucket_repository:         0.0000 sec
audit_specs:                        0.0002 sec
audit_reverse_migration:            0.0003 sec
audit_prefix_has_contents:          0.0010 sec
audit_lines:                        0.0011 sec
audit_formula_name:                 0.0029 sec
audit_deps:                         0.0038 sec
audit_file:                         0.0067 sec
audit_text:                         0.0227 sec
audit_installed:                    0.1935 sec
audit_revision_and_version_scheme:  12.6690 sec
```

I'm not confident this works as expected so would appreciate detailed review.

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -709,86 +709,74 @@ def audit_specs
     end
 
     def audit_revision_and_version_scheme
+      return if @new_formula
+      return unless @strict
```
Comment (Bo98) [inline Library/Homebrew/dev-cmd/audit.rb:713]: We don't run `--strict` in the CI but this particular audit is most useful when done in the CI.

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/audit.rb:713]: Yeh, I had in my head we did run `--strict` in CI but you're right, we don't. I think this probably warrants another flag e.g. `--slow`, `--git` or even `--ci`.

**Comment by MikeMcQuaid** [comment]
@Bo98 Have added a `--git` (PR to add it to test-bot incoming). Let me know what you think. I'm aware some tests are failing.

**Comment by Bo98** [comment]
Makes sense

**Pull request merged**